File: plotter.js
 var plotter = function(json_data){
	plot_data = {}; // To store data
	var plot_data_f = function(json_data){
		plot_data[["data"]] = json_data[["ans"]];
		plot_data[["labs"]] = json_data[["labs"]];
		plot_data[["ranges"]] = json_data[["ranges"]];
		len = json_data[["ans"]][["x"]].length;
			
	 	var data_scatter = [];
	 	for(var i=0; i<len; i++){
	 		data_scatter.push([ plot_data[["data"]][["x"]][[i]], plot_data[["data"]][["y"]][[i]] ]);
	 	}
	 	
	 	// plot data
	 	// Create SVG element
	 	var w = 500 // Width 
	 	var h = 500 // Height
		var svg = d3.select("body")
		            .append("svg")
		            .attr("width", w)
		            .attr("height", h);

		svg.selectAll("circle")
			.data(data_scatter)
			.enter()
			.append("circle")
			.attr("cx", function(d) {
			    return d[0];
			})
			.attr("cy", function(d) {
			    return d[1];
			})
			.attr("r", 3);
	}

	plot_data_f(json_data);
}

var loader = function(file_name){
	var json_data;
	// Load data from file
	d3.json(file_name, function(error, data) {
		if (error) {
			console.log(error);  //Log the error.
	    } else {
	        console.log(data);   //Log the data.
	    }
		json_data = data;
		plotter(json_data);
	});
}

File: R/plotter.R
#' Saves plot data to disk 
#' @param inp_data data to be plotted
#' @param graphType type of graph to be plotted
#' @param graphOptions options for the selected graph
#' @return nothing, data is stored to disk
#' @import htmlwidgets
#' @export
plotter <- function(inp_data = NULL, graphType = NULL,
                    graphOptions = list(), directory = "~/test_dir"){
  if(is.null(inp_data)){
    stop("No input data provided")
  }
  
  # Defalut graph type is scatter plot
  if(is.null(graphType)){
    graphType <- "scatter"
  }
  # Default columns to select are x & y
  if(length(graphOptions) < 1){
    graphOptions <- list(x="x",
                         y="y")
  }

  # Restructure data if required
  # Note: Does nothing yet
  data_clean <- cleanData(inp_data)
  # Get detailed info on data
  data_build <- editData(data_clean, graphType, graphOptions)
  saveData(data_build, directory)
}

cleanData <- function(inp_data){
  inp_data
}

editData <- function(data_clean, graphType, graphOptions){
  data_clean
  # Select only columns we need to save
  ans <- data.frame(x = data_clean[[graphOptions$x]],
                    y = data_clean[[graphOptions$y]])
  # Axis labels
  if(is.null(graphOptions$xlab)){
    graphOptions$xlab <- graphOptions$x
  }
  if(is.null(graphOptions$ylab)){
    graphOptions$ylab <- graphOptions$y
  }
  
  labs <- list()
  labs$xlab <- graphOptions$xlab
  labs$ylab <- graphOptions$ylab
  
  # Axis ranges
  ranges <- list()
  ranges$xrange <- range(ans$x, na.rm = T)
  ranges$yrange <- range(ans$y, na.rm = T)
  
  list(ans = ans,
       labs = labs,
       ranges = ranges)
}

#' Saves plot data to disk
#' @import htmlwidgets
#' @export
saveData <- function(data_build, directory){
  tsv.name <- sprintf("data_plot.tsv")
  tsv.path <- file.path(tools::file_path_as_absolute(directory),
                        tsv.name)
  write.table(data_build$ans, tsv.path,
              quote = FALSE, row.names = FALSE,
              sep = "\t")
  data_build$ans <- NULL
  data_build$path <- tsv.name
  # Create htmlwidget
  # Note: Pass the data as saved in plot.json
  widget <- htmlwidgets::createWidget(name = "plotterWidget",
                                      x = data_build,
                                      width = 400,
                                      height = 400,
                                      package = 'plotteR')
  # savewidget
  index.path <- file.path(tools::file_path_as_absolute(directory),
                        "index.html")
  htmlwidgets::saveWidget(widget, index.path)
}

printMessage <- function(){
  message("Plot saved to disk, opening window")
}
